lib/src/widgets/board.dart

```diff
@@ -453,6 +450,7 @@ class _BoardState extends State<Board> {
       _shouldDeselectOnTapUp = selected == squareId;
       setState(() {
         selected = squareId;
+        _premove = null;
```

Comment:
This is the only real change. Rest are just minor refactor to improve readability.